rust_src/src/fns.rs

```diff
@@ -30,4 +30,44 @@ fn featurep(feature: LispObject, subfeature: LispObject) -> LispObject {
     }
 }
 
+/// Announce that FEATURE is a feature of the current Emacs.
+/// The optional argument SUBFEATURES should be a list of symbols listing
+/// particular subfeatures supported in this version of FEATURE.
+#[lisp_fn(min = "1")]
+fn provide(feature: LispObject, subfeature: LispObject) -> LispObject {
+    feature.as_symbol_or_error();
+    if !subfeature.is_list() {
+        wrong_type!(Qlistp, subfeature)
+    }
+    unsafe {
+        if LispObject::from(Vautoload_queue).is_not_nil() {
+            Vautoload_queue = Fcons(
+                Fcons(LispObject::from_fixnum(0).to_raw(), globals.f_Vfeatures),
+                Vautoload_queue,
+            )
+        }
+    }
+    if memq(feature, LispObject::from(unsafe { globals.f_Vfeatures })).is_nil() {
+        unsafe {
+            globals.f_Vfeatures = Fcons(feature.to_raw(), globals.f_Vfeatures);
+        }
+    }
+    if subfeature.is_not_nil() {
+        put(feature, LispObject::from(Qsubfeatures), subfeature);
+    }
+    unsafe {
+        globals.f_Vcurrent_load_list = Fcons(
```

Comment:
This loses the name `LOADHIST_ATTACH`. Should we add a Rust function to keep the name alive?

Reply:
I've thought a while about that one. It's only used in a few places and I think it's easy to read with the formatting.